src/averageAges.js

```diff
@@ -54,8 +84,27 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childs = (!onlyWithSon) ? people : filterSex(people, 'm');
+
+  let countPeople = 0;
+
+  const ages = childs.reduce(
+    (sum, person) => {
+      const motherName = people.find(human => person.mother === human.name);
+      let iterSum = sum;
+
+      if (motherName !== undefined) {
+        countPeople++;
+        iterSum += person.born - motherName.born;
+      }
```

Comment:
and here, instead of summing ages differences and counting people , get an array of differences and send into calcAverage(ages differences)